Zend/zend_compile.c

```diff
@@ -11608,6 +11608,12 @@ static void zend_compile_expr_inner(znode *result, zend_ast *ast) /* {{{ */
 			zend_compile_match(result, ast);
 			return;
 		default:
+			zend_error(
+					E_ERROR,
+					"Unknown AST node kind %d. Expression: \"%s\".",
+					ast->kind,
+					ZSTR_VAL(zend_ast_export("", ast, ""))
```

Comment:
This means that `zend_ast_export` must be functional, which is not given when the AST is broken so that PHP is unable to handle it.

The included gdb script is capable of pretty-printing the AST. I suggest debugging this situation with gdb.